pyproject.toml

```diff
@@ -47,6 +49,16 @@ warn_return_any = true
 warn_redundant_casts = true
 namespace_packages = true
 
+[tool.pyright]
+include = ["crescent", "examples"]
+exclude = ["tests"]
+ignore = ["examples/basic/basic.py", "examples/basic/autocomplete.py"]
+pythonVersion = "3.8"
+typeCheckingMode = "strict"
+reportPrivateUsage = false
+reportImportCycles = false
+reportIncompatibleMethodOverride = false
```

Comment:
explain

Reply:
- reportPrivateUsage is disabled because crescent uses private stuff outside their defined module/class in other files
- reportImportCycles is disabled because I'm not sure it accounts for `typing.TYPE_CHECKING` based on ashlen's PR for hikari. We can take a look at this again to make sure if we need to change anything
- reportIncompatibleMethodOverride is disabled because it's based off of the parameters of a subclass and how the signature is extended: https://github.com/microsoft/pyright/issues/867

Reply:
> reportIncompatibleMethodOverride is disabled because it's based off of the parameters of a subclass and how the signature is extended: https://github.com/microsoft/pyright/issues/867

That issue appears to have been resolved.

Reply:
Why don't we just fix the order?

Reply:
Nope, in the above case I think it's because of attrs:
![image](https://user-images.githubusercontent.com/81032623/175041164-1c645aca-ed0e-453c-b19d-9f7c7a81ea76.png)

Reply:
Also why is that a classmethod

Reply:
I reverted the change to staticmethod, that was wrong, my bad.

Reply:
I don't think we can disable the incompatible method overrides, it doesn't particularly matter that much. While we're here though, should some things be reordered to match hikari?

Reply:
I dont think you should do that. Its too much for one PR.

Reply:
Got it!